mtp_bank_admin/apps/bank_admin/tests/test_statement.py

```diff
@@ -156,6 +161,26 @@ def test_reconciles_date(self, mock_api_client):
 
         conn.reconcile.post.assert_called_with({'date': today.isoformat()})
 
+    def test_batch_creation_batched_correctly(self, mock_api_client):
+        _, test_data = mock_test_transactions(mock_api_client, count=1050)
+        mock_balance(mock_api_client)
+        batch_conn = mock_batch(self, mock_api_client, test_data)
+
+        _, bai2_file = generate_bank_statement(self.get_request(),
+                                               datetime.now().date())
+        self.assertTrue(batch_conn.post.side_effect.called)
+
+        t_ids = [t['id'] for t in test_data['results']]
+        batch_conn.assert_called_with(1)
+        expected_calls = []
+        for i in range(1, math.ceil(len(t_ids)/settings.REQUEST_PAGE_SIZE)):
+            expected_calls.append(mock.call({'transactions': t_ids[
+                i*settings.REQUEST_PAGE_SIZE:(i+1)*settings.REQUEST_PAGE_SIZE
```

Comment:
This slice is a bit dense; might be clearer what the intent is if it's broken out into variables?